aspnet/mvc/models/validation.rst

```diff
@@ -1,8 +1,196 @@
+
+Alternatively, this same code could be placed in the model instead by implementing ``IValidatableObject`` which contains a ``Validate`` method. This works well for one-off validations like in the sample.
+
+ .. literalinclude:: validation/sample/MovieIValidatable.cs
+   :language: c#
+   :lines: 35-48
+   :dedent: 8
+  
+Client side validation
+----------------------
+Client side validation is a great convenience for users. It saves time they would otherwise spend waiting for a round trip to the server. In business terms, even a few seconds multiplied hundreds of times each day adds up to be a lot of time, expense, and frustration. Straightforward and immediate validation enables users to work more efficiently and produce better quality input and output. 
+
+You must have a view with the proper JavaScript script references in place for client side validation to work. 
+
+.. code-block:: html
+
+  <script src="~/lib/jquery/jquery.js"></script>
+  <script src="~/lib/jquery-validation/dist/jquery.validate.js"></script>
+  <script src="~/lib/jquery-validation-unobtrusive/jquery.validate.unobtrusive.js"></script>
+
+MVC uses data annotations in addition to type metadata from model properties to validate data and display any error messages using JavaScript. When you use MVC to render form elements from a model using `Tag Helpers <https://docs.asp.net/en/latest/mvc/views/tag-helpers/index.html>`_ or `HTML helpers <https://docs.asp.net/en/latest/mvc/views/html-helpers.html>`_ it will add HTML 5 `data- attributes. <http://w3c.github.io/html/dom.html#embedding-custom-non-visible-data-with-the-data-attributes>`_ in the form elements that need validation, as shown below. MVC generates the ``data-`` attributes for both built-in and custom attributes. The ``data-val-required`` attribute below contains an error message to display if the user doesn't fill in the release date field, and that message displays in the accompanying ``<span>`` element. You can display validation errors on the client using the relevant tag helpers as shown here:
+
+.. code-block:: html
+ :emphasize-lines: 5, 10
+
+  <form asp-action="Create">
+    <div class="form-horizontal">
+      <h4>Movie</h4>
+      <hr />
+      <div asp-validation-summary="ValidationSummary.ModelOnly" class="text-danger"></div>
```

Comment:
make sure all internal links are of the :doc: format and not full external link format.